tests/tests/tx_subscription.rs

```diff
@@ -0,0 +1,448 @@
+        .map(|secret| Input::owner(&secret.public_key()))
+        .collect();
+
+    // Create producer and validator with default configuration (subscribe_to_transactions=true)
+    let Nodes {
+        producers,
+        validators,
+        bootstrap_nodes: _dont_drop,
+    } = make_nodes(
+        pub_keys
+            .iter()
+            .map(|pub_key| Some(BootstrapSetup::new(*pub_key))),
+        secrets.clone().into_iter().enumerate().map(|(i, secret)| {
+            Some(
+                ProducerSetup::new(secret)
+                    .with_txs(num_txs)
+                    .with_name(format!("{}:producer", pub_keys[i])),
+            )
+        }),
+        pub_keys.iter().flat_map(|pub_key| {
+            (0..num_validators).map(move |i| {
+                Some(ValidatorSetup::new(*pub_key).with_name(format!("{pub_key}:{i}")))
+            })
+        }),
+        None,
+    )
+    .await;
+
+    // Verify the configuration was applied correctly
+    if let Some(p2p) = &producers[0].node.shared.config.p2p {
+        eprintln!(
+            "Default Producer subscribe_to_transactions: {}",
+            p2p.subscribe_to_transactions
+        );
+        assert!(
+            p2p.subscribe_to_transactions,
+            "Producer should have subscribe_to_transactions=true by default"
+        );
+    }
+    if let Some(p2p) = &validators[0].node.shared.config.p2p {
+        eprintln!(
+            "Default Validator subscribe_to_transactions: {}",
+            p2p.subscribe_to_transactions
+        );
+        assert!(
+            p2p.subscribe_to_transactions,
+            "Validator should have subscribe_to_transactions=true by default"
+        );
+    }
+
+    let client_one = FuelClient::from(producers[0].node.bound_address);
+    let client_two = FuelClient::from(validators[0].node.bound_address);
+
+    let (tx_id, _) = producers[0]
+        .insert_txs()
+        .await
+        .into_iter()
+        .next()
+        .expect("Producer is initialized with one transaction");
+
+    // Wait for the transaction to be committed on the producer node
+    let _ = client_one.await_transaction_commit(&tx_id).await.unwrap();
+
+    // Give some time for the transaction to be gossiped
+    tokio::time::sleep(Duration::from_secs(2)).await;
+
+    // Try to subscribe to the transaction on the validator node, this should work since gossip is enabled by default
+    let mut client_two_subscription = client_two
+        .subscribe_transaction_status(&tx_id)
+        .await
+        .expect("Should be able to subscribe for events");
+
+    // The transaction should be available on the validator node
+    tokio::time::timeout(Duration::from_secs(10), client_two_subscription.next())
+        .await
+        .expect("Should await transaction notification in time");
+
+    let response = client_two.transaction(&tx_id).await.unwrap();
+    assert!(
+        response.is_some(),
+        "Transaction should have been gossiped to validator"
+    );
```

Comment:
You can use `client_two` to commit any transaction to it, if gossiping is enabled, `client_one`(a.k.a. block prodcuer) will include it.

It will be a proof that gossiping enabled.